### ISSUE TITLE
new with on stage is even better than which with two stage

### ISSUE THREAD

**Issue by lucaslu1987** [MarekKowalski/DeepAlignmentNetwork#24]
Hi Marek,
i run  DANtest.py  and want to know the performance of different stage. However, the result of the network with one stage is even better. The data set is initialized as the paper said.below is the test result.(i used center distance)
**OneStage**
Processing common subset of the 300W public test set (test sets of LFPW and HELEN)
Average error: 0.428566349505
Processing challenging subset of the 300W public test set (IBUG dataset)
Average error: 0.598229653794
Showing results for the entire 300W pulic test set (IBUG dataset, test sets of LFPW and HELEN
Average error: 0.461809522334
**TwoStage**
Processing common subset of the 300W public test set (test sets of LFPW and HELEN)
Average error: 0.439570144885
Processing challenging subset of the 300W public test set (IBUG dataset)
Average error: 0.616254202783
Showing results for the entire 300W pulic test set (IBUG dataset, test sets of LFPW and HELEN
Average error: 0.474188937071

**Comment by mariolew** [comment]
@MarekKowalski Hi, in the training phase, did you hold the learning rate fixed to be 0.001?  If I use learning rate 0.001, then the loss in stage 2 cannot drop.

**Comment by MarekKowalski** [comment]
Yes, the learning rate was kept the same. When you are training the second stage are you updating the parameters for both stages or just stage 2? 

In my experience, if you are updating parameters of both stages at the same time, the error takes much longer to drop.

**Comment by mariolew** [comment]
@MarekKowalski  Actually, I only update parameters for stage 2, but I can see error drop only when I lower down the learning rate.

**Comment by MarekKowalski** [comment]
That's surprising, for how long are you trying to see if the error starts droping? Also, is that in your implementation or in the one in this repo?

**Comment by mariolew** [comment]
@MarekKowalski  I mean in my implementation, I only observe error drop with my learning rate lower down. And, someone has found almost the same problem in another tensorflow implementation. I've double checked the functionality of the self defined layer and found no problem, I don't know if the Adam is implemented differently in theano, or Batch Norm is implemented differently.

**Comment by zjjMaiMai** [comment]
@MarekKowalski @mariolew  Hi
In my implementation on tensorflow, I have not found this problem.
i use a private dataset , and epoch is [15,45]

![image](https://user-images.githubusercontent.com/13913992/39177200-34340ba6-47e1-11e8-9a8d-1ac94c2f36b3.png)

**Comment by jnulzl** [comment]
i want to know if both stages share parameters in the same layer?

**Comment by MarekKowalski** [comment]
No the parameter values are not shared between the stages.

Marek
